app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt

```diff
@@ -0,0 +1,140 @@
+package com.github.sdpteam15.polyevents.model.database.remote.objects
+
+import com.github.sdpteam15.polyevents.model.database.remote.DatabaseConstant
+import com.github.sdpteam15.polyevents.model.database.remote.DatabaseConstant.CollectionConstant.EVENT_COLLECTION
+import com.github.sdpteam15.polyevents.model.database.remote.DatabaseConstant.CollectionConstant.RATING_COLLECTION
+import com.github.sdpteam15.polyevents.model.database.remote.DatabaseInterface
+import com.github.sdpteam15.polyevents.model.database.remote.Matcher
+import com.github.sdpteam15.polyevents.model.database.remote.adapter.EventAdapter
+import com.github.sdpteam15.polyevents.model.database.remote.adapter.RatingAdapter
+import com.github.sdpteam15.polyevents.model.entity.Event
+import com.github.sdpteam15.polyevents.model.entity.Rating
+import com.github.sdpteam15.polyevents.model.entity.UserProfile
+import com.github.sdpteam15.polyevents.model.observable.Observable
+import com.github.sdpteam15.polyevents.model.observable.ObservableList
+
+const val TAG = "EventDatabase"
+
+class EventDatabase(private val db: DatabaseInterface) : EventDatabaseInterface {
+    override fun createEvent(event: Event, userAccess: UserProfile?): Observable<Boolean> =
+        db.addEntity(event, EVENT_COLLECTION, EventAdapter)
+
+    override fun updateEvents(event: Event, userAccess: UserProfile?): Observable<Boolean> =
+        db.setEntity(event, event.eventId!!, EVENT_COLLECTION)
+
+    override fun getEventFromId(
+        id: String,
+        returnEvent: Observable<Event>,
+        userAccess: UserProfile?
+    ): Observable<Boolean> =
+        db.getEntity(returnEvent, id, EVENT_COLLECTION, EventAdapter)
+
+    override fun getEvents(
+        matcher: Matcher?,
+        limit: Long?,
+        eventList: ObservableList<Event>,
+        userAccess: UserProfile?
+    ): Observable<Boolean> =
+        db.getListEntity(
+            eventList,
+            null,
+            {
+                if (matcher != null) {
+                    if (limit != null)
+                        matcher.match(it).limit(limit)
+                    else
+                        matcher.match(it)
+                } else {
+                    if (limit != null)
+                        it.limit(limit)
+                    else
+                        it
+                }
```

Comment:
Now that I see it we could probably replace this by:

```
var query = it
if (matcher != null) query = matcher.match(it)
if (limit != null) query = query.limit(limit)
query
```

Reply:
I totally agree, it has been changed in the last version !